### PR TITLE
WIP: re-export localindexes [ci skip]

### PR DESCRIPTION
**Pull request by timholy**
Ref https://github.com/JuliaLang/julia/pull/12412#discussion_r35987298

### CODE REVIEW

**Comment by tkelman** [comment]
removing too many exports isn't necessarily a bad thing...

**Comment by timholy** [comment]
Sure, if we want to leave this un-exported, that would be fine. But since it seems to have been an accident, I thought I'd raise it for discussion. CC @amitmurthy.

**Comment by amitmurthy** [comment]
I am for re-exporting, now that SharedArrays are not being marked as experimental and because this is a useful api for work partitioning.

**Comment by timholy** [comment]
@one-more-minute, I just rebased this attempting to work with the new documentation system. (The "Blah blah" part of this commit is obviously something I might have to reevaluate, so consider this a WIP :smile:.) 

Unfortunately, after `make && make -C doc html` I get this:
```
help?> localindexes
search: localindexes


julia>
```
and this:
image

Can you provide a bit more guidance about how we're supposed to proceed? I've read both http://docs.julialang.org/en/latest/manual/documentation/ and https://groups.google.com/d/msg/julia-users/eI-mZe2zA-c/kmc7RdPPAQAJ, is there some other place I should be looking?

**Comment by one-more-minute** [comment]
Not sure if the doc stuff will run as part of make, but you can force it with `julia doc/genstdlib.jl`. The other thing to be wary of is that we don't completely support documenting things before the doc system is loaded, which looks like it might be the case here, so you could try putting the docstring in `base/docs/basedocs.jl` to see if that helps.

**Comment by timholy** [comment]
Moving to `base/docs/basedocs.jl` made `?localindexes` work, thanks!

But no dice on the RST documentation (still "Blah blah"). I did this order:
```sh
make
./julia doc/genstdlib.jl
make -C doc html
firefox doc/_build/html/index.html &
```

**Comment by one-more-minute** [comment]
Ah, I forgot – since your docstring is in Markdown it needs #12435 in order to output to RST. If you leave it as it is I'll make sure that gets fixed as part of that PR.

**Comment by tkelman** [comment]
Would it help to just push the RST rendering backend part of that PR (and a test) to master? Rendering currently-markdown docstrings into RST is something people are going to want to do now, but converting all of the RST in helpdb into markdown might take a while to get totally right.

**Comment by one-more-minute** [comment]
Yeah, that makes sense, I went ahead and pushed that commit, so that should make things easier for this PR.

**Comment by timholy** [comment]
I rebased on top of aff076d7860ca1b23080c38b1933d42efaed9ddd, but without moving the docstring to `base/docs/basedocs.jl`, I'm still not getting anything for either version (REPL or RST docs). Should I just leave this here and let you play with it?

File: doc/stdlib/parallel.rst [hunk 1]
```diff
@@ -703,6 +703,10 @@ Shared Arrays (Experimental, UNIX-only feature)
    Returns the index of the current worker into the ``pids`` vector, i.e., the list of workers mapping
    the SharedArray
 
+.. function:: localindexes(S::SharedArray)
+
+   Blah blah
```
Comment (StefanKarpinski) [inline doc/stdlib/parallel.rst]: ??

Comment (timholy) [inline doc/stdlib/parallel.rst]: This was me testing the new documentation system---see the history in #12423. This will not be part of the final commit.

**Comment by timholy** [comment]
Subsumed in 9c595c38

**Comment by tkelman** [comment]
The doc addition for it wasn't though?

**Comment by timholy** [comment]
Good point. Leaving this open, short on time right now.

**Comment by timholy** [comment]
Superseded by #14532.

**Pull request closed without merging**